astropy/units/physical.py

```diff
@@ -338,11 +338,10 @@ class PhysicalType:
 
     def __init__(self, unit: core.UnitBase, physical_types: str | set[str]) -> None:
         self._unit = _replace_temperatures_with_kelvin(unit)
-        self._physical_type = _standardize_physical_type_names(physical_types)
-        self._physical_type_list = sorted(self._physical_type)
```

Comment:
I suspect that I originally separated the `list` and `set` because of code implemented elsewhere in the class that didn't make it into the final version of #11204.  Good catch on consolidating this!